### PR TITLE
Long due improvements

### LINKED ISSUE #30: Traceback for non-existing package
When I try to run `pyp2spec` for a nonexistent package, I get a traceback

```
$ pyp2spec nonexistingpackage 

Generating configuration file
Assuming 'nonexistingpackage' is a package name
Querying PyPI for package 'nonexistingpackage'
Assuming changelog --message=Initial package
Assuming --description=This is package 'nonexistingpackage' generated automatically by pyp2spec.
Traceback (most recent call last):
  File "/usr/bin/pyp2spec", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyp2spec/pyp2spec.py", line 63, in main
    config_file = create_config(options)
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyp2spec/pyp2conf.py", line 379, in create_config
    contents = create_config_contents(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyp2spec/pyp2conf.py", line 320, in create_config_contents
    summary = pkg.summary()
              ^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyp2spec/pyp2conf.py", line 161, in summary
    summary = self.package_data["info"]["summary"]
              ~~~~~~~~~~~~~~~~~^^^^^^^^
KeyError: 'info'
```

### PR DESCRIPTION
**Pull request by befeleme**
Each commit deals with a one issue:
- fixed #30
- config filenames are newly prepeneded with `_` to enable smooth autocompletion with one TAB press when using RPM tools with specfiles 
- use rpmautospec macros and remove all the things related to defining custom changelogs and releases

### CODE REVIEW

File: pyp2spec/pyp2conf.py [hunk 1]
```diff
@@ -46,12 +43,13 @@ def get_package_metadata(self, *,session=None):
         pkg_index = f"https://pypi.org/pypi/{self.package_name}/json"
         s = session or requests.Session()
         try:
-            response = s.get(pkg_index).json()
-        except json.decoder.JSONDecodeError:
+            response = s.get(pkg_index)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
```
Comment (hroncok) [inline pyp2spec/pyp2conf.py:48]: This seems rather unnatural to me. The raise_for_status method is designed to propagate the exception further up. If you just want to catch it right away, wouldn't it be easier to use `if not response.ok:`?

File: tests/test_pyp2conf.py [hunk 2]
```diff
@@ -264,3 +253,4 @@ def test_pypi_version_or_macro(pypi_version, pypi_version_macro):
     fake_pkg_data = {"info": {"version": pypi_version}}
     pkg = PypiPackage("_", package_metadata=fake_pkg_data)
     assert pkg.pypi_version_or_macro(pypi_version) == pypi_version_macro
+
```
Comment (hroncok) [inline tests/test_pyp2conf.py:256]: :roll_eyes:

**Comment by hroncok** [comment]
What motivated the _prefix change? Is there a discussion I can read? Or a precedence in another similar tool? E.g. the rpmlintrc files don't have this prefix and I never considered that a problem.

**Comment by hroncok** [comment]
I don't see that problem:

```
[python-pip (rawhide)]$ ls python-pip.*
python-pip.rpmlintrc  python-pip.spec
[python-pip (rawhide)]$ touch python-pip.conf
[python-pip (rawhide %)]$ ls python-pip.*
python-pip.conf  python-pip.rpmlintrc  python-pip.spec
[python-pip (rawhide %)]$ spectool -g p[TAB]
[python-pip (rawhide %)]$ spectool -g python-pip.spec 
```

**Comment by befeleme** [comment]
I see, this is an issue between bash and zsh completion, I just checked it. Me and Lumír are using zsh where the behaviour is different.

**Comment by hroncok** [comment]
spectool only has Bash completion file, not ZSH, see https://pagure.io/rpmdevtools/blob/main/f/rpmdevtools.bash-completion.in

Anyway, I don't want to block this, but the underscored config looks weird to me -- other tools don't follow this practice, so IMHO you are establishing a nonstandard and hence unexpected behavior.

**Comment by befeleme** [comment]
I value the feedback. I'm dropping the commit for now as I have no strong desire to create precedence that'll be weird to use among the known tools.

**Review by hroncok — APPROVED** [review]
I consider the remaining 2 commits non-controversial, well-scoped, and generally awesome :)

**Pull request merged**